src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts

```diff
@@ -57,7 +64,38 @@ export class CreateContactsFormComponent implements OnInit, OnDestroy {
     this.passLegalAddressFormGroup.emit(this.LegalAddressFormGroup);
   }
 
+  onSelectedCity(event: any, type: string): void {
+    if (type === 'legal') {
+      this.cityLegal = event.name;
+      this.LegalAddressFormGroup.get('city').reset();
+      this.LegalAddressFormGroup.get('city').setValue(event.name);
+    } else if (type === 'actual') {
+      this.cityActual = event.name;
+      this.ActualAddressFormGroup.get('city').reset();
+      this.ActualAddressFormGroup.get('city').setValue(event.name);
```

Comment:
split it to two different methods, because they have nothing in common
let have onSelectActualCity that will handle only actual control
and onSelectLegalCity that will handle only legal control